src/main/java/com/couchbase/lite/store/SQLiteStore.java

```diff
@@ -289,6 +289,9 @@ else if (e.getCode() == SQLException.SQLITE_ENCRYPTION_NOTAVAILABLE)
             }
 
             if (isNew)
+                setInfo("pruned", "true"); // See -compact: for explanation
+
+            if (isNew)
```

Comment:
Thank you very much for finding this mistake. `optimizeSQLIndexes()` was never executed before.....